### PR TITLE
Memoize reached quorum

### PR DESCRIPTION
**Pull request by DarksightKellar**
`votesSummary` is apparently something that _might_ be undefined, as implied by existing undefined checks. I've added those checks to `reachedQuorum` as well. 

_Might possibly address:_ #1724 

Edit for context:
I did search for references in the codebase for a `.no` access attempt, which led me to these files. Untested, obviously, esp since the issue itself isn't consistent. Not linking this PR (or assigning myself to the issue) as I'm not sure if it for sure fixes it. Not sure what to do here actually, in terms of issue management.

### CODE REVIEW

File: src/utils/hierarchy.ts [hunk 1]
```diff
@@ -1,18 +1,7 @@
 import { DAO } from '../../.graphclient';
 import { Node } from '../types';
 
-type DAOQueryNode = Pick<
```
Comment (DarksightKellar) [inline src/utils/hierarchy.ts:4]: Should this code even be in the reskin branch?

Comment (adamgall) [reply src/utils/hierarchy.ts:4]: It makes sense to me that it's still here -- when I had removed it on the call with us on Tuesday, it was during a merge conflict resolution session. Yesterday I "undid" that merge, so this code got put back.

**Comment by adamgall** [comment]
@DarksightKellar why do you think this would address #1716 ?

**Comment by DarksightKellar** [comment]
> @DarksightKellar why do you think this would address #1716 ?

I goofed up. I've updated the description to the correct issue   I meant to ref

**Review by adamgall — CHANGES_REQUESTED** [review]
Chatted with the team about this.

The issue is _very likely_ that for a split second there are some data loading issues, and the app thinks the given dao/proposal is a multisig, when in fact it's an Azorius, and the currently rendered QuorumBadge goes to re-render, and we force its type to AzoriusProposal, but it isn't at that split second, and the app crashes.

So,

for now, we can just bandaid fix this. Move the bandaid up as far as possible, maybe likely just up into the ProposalCard.

The better fix is to rearchitect and refactor our typing system and data loading system. But that's a project for another day.

**Comment by DarksightKellar** [comment]
@adamgall Pushed an update. Because `QuorumBadge` now requires an `AzoriusProposal`, the undefined checks are no longer necessary, so they've been removed. 

The `as AzoriusProposal` has simply moved further up to the `ProposalCard` though. But that is more confident type assertion. Any higher up and the proposal can only be a `FractalProposal` until context determines what it actually is.

Further thoughts: I dunno that much can be done about rearchitecting the typing system (in this context) tbh. Probably makes more sense to focus (when we do get to it) on figuring out the data loading flow that's causing this state disconnect.

File: src/components/Proposals/ProposalCard/ProposalCard.tsx [hunk 2]
```diff
@@ -28,6 +33,7 @@ function ProposalCard({ proposal }: { proposal: FractalProposal }) {
   );
 
   const isSnapshotProposal = !!(proposal as SnapshotProposal).snapshotProposalId;
+  const isAzoriusProposal = !!(proposal as AzoriusProposal).votesSummary;
```
[the remark this answers is not in the document]
Comment (DarksightKellar) [reply src/components/Proposals/ProposalCard/ProposalCard.tsx:36]: The thing is, it's not absolutely certain that `proposal` _will be_ an `AzoriusProposal`. 

There's no point in making that assumption early on, only to discover that `azoriusProposal` doesn't in fact have a `.votesSummary` on attempting to pass it to `QuorumBadge` (so it's not actually an `AzoriusProposal`), and then just not using `azoriusProposal` at all.

[the remark this answers is not in the document]
Comment (adamgall) [reply src/components/Proposals/ProposalCard/ProposalCard.tsx:36]: @mudrila i'm confused about what you're saying. Where do you see `proposal as AzoriusProposal` twice?

File: src/components/Proposals/ProposalSummary.tsx [hunk 3]
```diff
@@ -76,6 +74,17 @@ export default function ProposalSummary({ proposal }: { proposal: AzoriusProposa
...
+
+  const reachedQuorum = useMemo(
+    () =>
+      isERC721
+        ? totalVotesCasted - no
+        : votesToken
+          ? (totalVotesCasted - no) / votesTokenDecimalsDenominator
+          : 0n,
+    [isERC721, no, totalVotesCasted, votesToken, votesTokenDecimalsDenominator],
+  );
```
Comment (adamgall) [inline src/components/Proposals/ProposalSummary.tsx:86]: I like the code you wrote in QuorumBadge to turn these nested ternaries into various if blocks. Can you do that here too?

File: src/components/Proposals/ProposalSummary.tsx [hunk 4]
```diff
@@ -21,14 +21,12 @@ import { ProposalAction } from './ProposalActions/ProposalAction';
...
-    deadlineMs,
-    proposer,
-    transactionHash,
-  } = proposal;
+  const { eventDate, startBlock, votesSummary, deadlineMs, proposer, transactionHash } = proposal;
+
+  const yes = votesSummary?.yes || 0n;
+  const no = votesSummary?.no || 0n;
+  const abstain = votesSummary?.abstain || 0n;
+
```
Comment (adamgall) [inline src/components/Proposals/ProposalSummary.tsx:29]: For the sake of trying to isolate where the bug actually appears, can you please undo this?

**Review by adamgall — CHANGES_REQUESTED** [review]

**Review by adamgall — APPROVED** [review]

File: src/components/Proposals/ProposalCard/ProposalCard.tsx (hunk 2 above)
[the remark this answers is not in the document]
Comment (DarksightKellar) [reply src/components/Proposals/ProposalCard/ProposalCard.tsx:36]: It's further down, passing proposal as a prop to QuoromBadge

[the remark this answers is not in the document]
Comment (adamgall) [reply src/components/Proposals/ProposalCard/ProposalCard.tsx:36]: Ah ok, thanks. Regardless, I implemented the changes I requested in my review and merged this branch already!

**Pull request merged**